markdown/extensions/codehilite.py

```diff
@@ -75,7 +79,8 @@ class CodeHilite(object):
 
     def __init__(self, src=None, linenums=None, guess_lang=True,
                  css_class="codehilite", lang=None, style='default',
-                 noclasses=False, tab_length=4, hl_lines=None, use_pygments=True):
+                 noclasses=False, tab_length=4, use_pygments=True,
+                 lexer_options={}, formatter_options={}):
```

Comment:
I'm thinking we should just do `**options` here. That does two things:

1. It avoids the extra level of nesting when defining the configs.
2. It avoids separating lexer and formatter options.

As it stands now, Pygments uses `**options` in their lexer and formatter classes, and ignores any unknown option keywords. So as long as there are no naming conflicts between lexer and formatter options, we could pass all options to both the lexer and formatter. This has the added benefit of making it easier to define options from within documents (see #816).